### DISCUSSION TITLE
ReferenceError: __dirname is not defined & middleware issue when deploying next-auth on Vercel

### DISCUSSION THREAD

**Discussion by walter-kai** [vercel/community#6169] [Help]
### Summary

I made a copy of the Next.js Vercel tutorial and so I'm now using my own copy at this repo:
https://github.com/yaozakai/nextjs-dashboard

The deployment successfully completes, but the page displayed is showing "**The Edge Function** has crashed"

as shown here: https://nextjs-dashboard-git-main-walts-projects-134c10a9.vercel.app/

Vercel logs are showing errors that keep popping up:

`[GET] [middleware: "middleware"] /favicon.ico reason=EDGE_FUNCTION_INVOCATION_FAILED, status=500, user_error=true`

and

`ReferenceError: __dirname is not defined
    at (node_modules/next/dist/compiled/ua-parser-js/ua-parser.js:1:17315)
    at (node_modules/next/dist/compiled/ua-parser-js/ua-parser.js:1:17444)
    at (node_modules/next/dist/server/web/spec-extension/user-agent.js:27:59)
    at (node_modules/next/server.js:8:0)
    at (node_modules/next-auth/lib/index.js:3:24)
    at (node_modules/next-auth/index.js:52:0)`

I've tried using WebPacks, and Babel to help fix the __dirname, but I think I'm going the wrong direction as I shouldn't need to manually replace __dirname within ua-parser.js.

Is my configuration incorrect?

### Example

nextjs-dashboard-git-main-walts-projects-134c10a9.vercel.app

### Steps to Reproduce

clone and deploy [](https://github.com/yaozakai/nextjs-dashboard) to vercel, you will see the error screen and the logs will show the same errors as mentioned above.

**Comment by GauravBurande** [comment]
I got this error too! I think middleware is still unstable for app router

**Reply by walter-kai** [reply]
I think you're right about the middleware, because there are some issues with trying to use Vercel as a Node.js server it seems as it's meant for a serverless environment, I assume.

my solution: don't follow the tutorial at next.js tutorial (sorry Next.js) and use the Next-Auth template offered in Vercel. I used that and modified it to my liking and now it works exactly as expected.
